doc/developer/platform/ux.md

```diff
@@ -1,20 +1,726 @@
-A CLUSTER contains indexes and is the unit within which it is possible to share indexes.
+Terms in this document that represent key user-facing concepts are introduced
+with *italics*. The underlying components may use different terminology
+internally, but any user-facing interfaces should use the term specified in this
+document. For example, the COMPUTE layer is free to use the term COMPUTE
+INSTANCE internally, but the user-facing documentation, web UI, and SQL API
+should all consistently use the term *cluster*.
 
-Within each REGION there may be multiple TIMELINEs, which correspond roughly to coordinator timelines.
-A TIMELINE contains collections, and is the unit within which it is possible to query across multiple collections
+## Overview
 
-The main restrictions we plan to impose on users is that their access to data is always associated with a REGION, CLUSTER, and TIMELINE, and attempts to access data across these may result in poorer performance, or explicit query rejection.
+Depicted below are the tangible concepts that Materialize Platform exposes to
+users.
 
-There is some work to do to present these concepts to users, but prior work exists with e.g. SQL `database`s for TIMELINE and Snowflake's `USE WAREHOUSE` command to pilot CLUSTER.
+![Diagram of user-facing concepts](./assets/user-concepts.png)
+
+An *account* represents a customer of Materialize Platform. An account has
+*users* and manages *billing*.
+
+An *region* is a contains the account's infrastructure in a specific region
+of a cloud provider (e.g., AWS us-east-1). The region boundary is strong:
+besides users, no objects are shared between regions.
+
+A *cluster* is the user-facing name for a [COMPUTE
+INSTANCE](overview.md#COMPUTE). A region can contain any number of clusters.
+
+A region contains a standard SQL catalog, which consists of *databases*, which
+each contain *schemas*, which each contain *database objects*. A database object
+is one of the following:
+
+  * a *source*, which imports data into the region;
+  * a *sink*, which exports data out of the region;
+  * a *table*, which stores data directly in the region;
+  * a *view*, which describes a transformation of sources, tables, and other
+    views;
+  * an *index*, which schedules the persistent computation of a source or
+    view on a cluster;
+  * a *secret*, which securely stores credentials for external systems.
+
+Note that sources, sinks, tables, views, and secrets are shared across the
+entire cluster, while indexes are only shared within a cluster.
+
+A user's access to a region is regulated by *roles*. A role is unique to an
+region and defines what users are allowed to assume the role and which database
+objects are readable and writable.
+
+Each region contains a single [STORAGE INSTANCE](overview.md#STORAGE). Users are
+not directly aware of this storage instance, but any persistent state in the
+region (e.g., table data) is managed by the regions's storage instance.
```

Comment:
No strong feelings myself, but is this a hard requirement, or a ux feature of the cloud product? Like, someone says "I need two storage instances for compliance" we come back and say "you require a second account"?

Reply:
So I think this is exactly the difference between allowing at most one environment per region, and allowing any number of environments per region! Like if you mentally s/environment/STORAGE INSTANCE/, I think we are on the same page. There's some discussion in "environments vs regions" below. My contention is this: it's very easy to evolve Materialize Platform down to the road to allow people to create multiple environments per region. But if we _start_ with that feature, and it turns out to be a confusing layer of the hierarchy that no one ever really uses, much harder to take it away.